pkg/lvm/lvm2app.go

```diff
@@ -892,23 +574,112 @@ func ListVolumeGroupNames() ([]string, error) {
 // This does not normally scan for devices. To scan for devices, use
 // the `Scan()` function.
 func ListVolumeGroupUUIDs() ([]string, error) {
-	return defaultHandle.ListVolumeGroupUUIDs()
+	result := new(vgsOutput)
+	if err := run("vgs", result, "--options=vg_uuid"); err != nil {
+		return nil, err
+	}
+	var uuids []string
+	for _, report := range result.Report {
+		for _, vg := range report.Vg {
+			uuids = append(uuids, vg.UUID)
+		}
+	}
+	return uuids, nil
 }
 
 // CreatePhysicalVolume creates a physical volume of the given device.
 func CreatePhysicalVolume(dev string) (*PhysicalVolume, error) {
-	return defaultHandle.CreatePhysicalVolume(dev)
+	if err := run("pvcreate", nil, dev); err != nil {
+		return nil, fmt.Errorf("lvm: CreatePhysicalVolume: %v", err)
+	}
+	return &PhysicalVolume{dev}, nil
+}
+
+type pvsOutput struct {
+	Report []struct {
+		Pv []struct {
+			Name   string `json:"pv_name"`
+			VgName string `json:"vg_name"`
+		} `json:"pv"`
+	} `json:"report"`
 }
 
 // ListPhysicalVolumes lists all physical volumes.
 func ListPhysicalVolumes() ([]*PhysicalVolume, error) {
-	return defaultHandle.ListPhysicalVolumes()
+	result := new(pvsOutput)
+	if err := run("pvs", result); err != nil {
+		return nil, err
+	}
+	var pvs []*PhysicalVolume
+	for _, report := range result.Report {
+		for _, pv := range report.Pv {
+			pvs = append(pvs, &PhysicalVolume{pv.Name})
+		}
+	}
+	return pvs, nil
 }
 
 // LookupPhysicalVolume returns a physical volume with the given name.
 func LookupPhysicalVolume(name string) (*PhysicalVolume, error) {
-	return defaultHandle.LookupPhysicalVolume(name)
+	result := new(pvsOutput)
+	if err := run("pvs", result, "--options=pv_name", name); err != nil {
+		if IsPhysicalVolumeNotFound(err) {
+			return nil, ErrPhysicalVolumeNotFound
+		}
+		return nil, err
+	}
+	for _, report := range result.Report {
+		for _, pv := range report.Pv {
+			return &PhysicalVolume{pv.Name}, nil
+		}
+	}
+	return nil, ErrPhysicalVolumeNotFound
 }
 
 // Extent sizing for linear logical volumes:
 // https://github.com/Jajcus/lvm2/blob/266d6564d7a72fcff5b25367b7a95424ccf8089e/lib/metadata/metadata.c#L983
+
+func run(cmd string, v interface{}, extraArgs ...string) error {
+	var args []string
+	if v != nil {
+		args = append(args, "--reportformat=json")
```

Comment:
what's the min version requirement for LVM in order to get JSON report format? for example:
```sh
$ uname -a
Linux node-1 4.4.0-116-generic #140-Ubuntu SMP Mon Feb 12 21:23:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

$ dpkg -l|grep lvm
...
ii  lvm2                                2.02.133-1ubuntu10                         amd64        Linux Logical Volume Manager

$ sudo lvs --reportformat=json
lvs: unrecognized option '--reportformat=json'
  Error during parsing of command line.
```

we might want to include a min version check for LVM tooling at plugin startup and fail immediately if it's too old

Reply:
That's a really good point. I never considered that the CLI interface we rely on may have changed in living memory.